conseil-api-tests/src/main/scala/tech/cyptonomic/conseil/api/tests/ApiTestRun.scala

```diff
@@ -0,0 +1,289 @@
+      println(sendConseilRequest(httpClient, CONSEIL.withPath(Requests.getTezosAttributePath(entity))))
+    })
+  }
+
+  def testMetadata(httpClient: BlazeClientBuilder[IO], entityAttributes: Map[String, Array[String]] = Requests.TEZOS_ENTITY_ATTRIBUTES): Unit = {
+
+    entityAttributes.foreach(entity => {
+
+      entity._2.foreach(attribute => {
+
+        println("\n\n\nTesting Tezos " + entity._1 + " " + attribute + " Metadata\n")
+        val result: String = sendConseilRequest(httpClient, CONSEIL.withPath(Requests.getTezosMetadataPath(entity._1, attribute)))
+        println(result)
+
+      })
+
+    })
+
+  }
+
+  /**
+   * Test Conseil Attributes for each entity
+   * @param httpClient The BlazeClientBuilder[IO] object to make the requests to conseil
+   * @param entityAttributes A Map that has entities as keys, and an array of attributes as its values
+   */
+  def testAttributeData(httpClient: BlazeClientBuilder[IO], entityAttributes: Map[String, Array[String]] = Requests.TEZOS_ENTITY_ATTRIBUTES): Unit = {
+
+    entityAttributes.foreach(entity => {
+
+      val queryString: String =
+        """
+          |{
+          |     "fields": %FIELDS%,
+          |     "predicates": [],
+          |     "orderBy": [],
+          |     "aggregation": [],
+          |     "limit": 10
+          |}
+          |""".stripMargin.replace("%FIELDS%", entity._2.asJson.toString)
+
+      println("\n\n\nTesting Tezos " + entity._1 + " Query\n")
+      val result: String = sendConseilQuery(httpClient, CONSEIL.withPath(Requests.getTezosQueryPath(entity._1)), queryString)
+      println(result)
+
+      entity._2.foreach(attribute => {
+        validateJsonByKey(attribute, result)
+      })
+
+    })
+
+  }
+
+  /**
+   * Sends a query to make sure each attribute data is valid
+   * @param httpClient The BlazeClientBuilder[IO] object that sends the conseil requests
+   * @param entityAttributes A Map that has entities as keys, and an array of attributes as its values
+   */
+  def validateAttributeData(httpClient: BlazeClientBuilder[IO], entityAttributes: Map[String, Array[String]] = Requests.TEZOS_ENTITY_ATTRIBUTES): Unit = {
+
+    entityAttributes.foreach(entity => {
+
+      entity._2.foreach(attribute => {
+
+        val queryString: String =
+          """
+            |{
+            |     "field": ["%FIELD%"],
+            |     "predicates": [
+            |         {
+            |             "field": "%FIELD%",
+            |             "operation": "isnull",
+            |             "set": [""],
+            |             "inverse": true
+            |         }
+            |     ],
+            |     "orderBy": [],
+            |     "aggregation": [],
+            |     "limit": 1
+            |}
+            |""".stripMargin.replace("%FIELD%", attribute)
+
+        println("\n\n\nValidating Tezos " + entity._1 + " " + attribute + " Data\n")
+        var result: String = ""
+        try {
```

Comment:
I don't like try/catch here and in other cases. I would rather see handling errors with `.handleErrorWith(...)` method on `IO` and making calls asynchronous.

Reply:
Why should calls be asynchronous?

Reply:
Basically when you are using any Future/IO in Scala it is preferred to compose those effects and use `unsafeRunSync`/`unsafeRunAsync` just once in the main object.